lib/Tracy/DIPanel/DIPanel.php

```diff
@@ -0,0 +1,150 @@
+    $container_wrapper = ContainerWrapper::getInstance();
+    $reflection = new \ReflectionProperty(ContainerWrapper::class, $name);
+    $reflection->setAccessible(true);
+    $container = $reflection->getValue($container_wrapper);
+    $reflection = new \ReflectionProperty(get_class($container), 'services');
+    $reflection->setAccessible(true);
+    return [$reflection->getValue($container), $container->getDefinitions()];
+  }
+
+  /**
+   * For each service finds all of its arguments recursively and makes them an array
+   * @param string[] $services
+   * @return array
+   */
+  private function flattenArguments($services) {
+    $result = [];
+    foreach ($services as $service) {
+      $result[$service] = [];
+      $this->getAllArguments($service, $result[$service]);
+    }
+    return $result;
+  }
+
+  /**
+   * Find all argument of each service and adds them to $results array, repeats recursively
+   * @param string $service
+   * @param string[] $results
+   */
+  private function getAllArguments($service, &$results) {
+    if (array_key_exists($service, $this->definitions)) {
+      $arguments = $this->definitions[$service]->getArguments();
+      if (!empty($arguments)) {
+        foreach ($arguments as $argument) {
+          if (is_null($argument)) continue;
+          if ($argument instanceof TypedReference) {
+            $argumentName = $argument->getType();
+          } elseif ($argument instanceof Reference) {
+            $argumentName = (string)$argument;
+          } else {
+            continue;
+          }
+          $results[$argumentName] = $argumentName;
+          if ($argumentName !== 'MailPoet\DI\ContainerWrapper') {
+            $this->getAllArguments($argumentName, $results);
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * @param string $item
+   */
+  public function printItem($item) {
+    echo $item;
+    if (array_key_exists($item, $this->definitions)) {
+      $arguments = $this->definitions[$item]->getArguments();
+      if (!empty($arguments)) {
+        echo '<span class="tracy-toggle tracy-collapsed">...</span>';
+        echo '<div class="tracy-collapsed" style="padding-left: 10px">';
+        foreach ($arguments as $argument) {
+          if (is_null($argument)) {
+            echo 'NULL <br>';
+          } elseif ($argument instanceof TypedReference) {
+            $this->printItem($argument->getType());
+            echo '<br>';
+          } elseif ($argument instanceof Reference) {
+            $this->printItem((string)$argument);
+            echo '<br>';
+          } elseif (is_string($argument)) {
+            echo $argument;
+            echo '<br>';
+          }
+        }
+        echo '</div>';
+      }
+    }
+  }
+
+  /**
+   * @param string $item
+   */
+  public function printUsages($item) {
+    foreach ($this->arguments_flattened as $service => $arguments) {
+      if (array_key_exists($item, $arguments)) {
+        echo $service . '<br>';
```

Comment:
I think it would be nice to do this with toggle.
```
Toggle label: Used in XY services
- service 1
- service 2
...
```
With this change, service will need only a single line in panel.